src/decima/result/result.py

```diff
@@ -0,0 +1,331 @@
+from typing import List, Optional, Union
+from dataclasses import dataclass
+import torch
```

Comment:
I would suggest 

- renaming the module to `core` rather than `result`
- Separate the dataclasses and the result into two files.